contracts/Tokens/VTokens/VToken.sol

```diff
@@ -351,6 +344,168 @@ abstract contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
         protocolShareReserve = protcolShareReserve_;
     }
 
+    /**
+     * @notice Transfers the underlying asset to the specified address.
+     * @dev Can only be called by the Comptroller contract. This function performs the actual transfer of the underlying
+     *      asset by calling the `doTransferOut` internal function.
+     *      - The caller must be the Comptroller contract.
+     *      - If the `to` address is not the protocol share reserve, the flashLoanAmount is incremented by the amount transferred out.
+     * @param to The address to which the underlying asset is to be transferred.
+     * @param amount The amount of the underlying asset to transfer.
+     * @return balanceBeforeRepayFlashloan Cash balance after transfer out for comparison in flash loan verification.
+     * @custom:error InvalidComptroller is thrown if the caller is not the Comptroller.
+     * @custom:event Emits TransferOutUnderlying event on successful transfer of amount to receiver
+     */
+
+    function transferOutUnderlying(
+        address payable to,
+        uint256 amount
+    ) external nonReentrant returns (uint256 balanceBeforeRepayFlashloan) {
+        if (msg.sender != address(comptroller)) {
+            revert InvalidComptroller();
+        }
+
+        if (to != protocolShareReserve) {
+            flashLoanAmount += amount;
```

Comment:
Why `+=` instead of `=`? Do we want to support consecutive flash loans without repaying the previous one?

I know we have the `nonReentrant` check, but would it make sense to also revert the TX if `flashLoanAmount > 0`? I think it would cover a different scenario: a new flash loan without repaying the previous one

Reply:
01d36ee5ab694143c07c2d96d924cd2e91a7e9ad